### PR TITLE
Add comprehensive AI translation end-to-end tests

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

Adds a suite of end-to-end tests for the AI translation feature, covering real OpenAI API integration, offline queueing, translation direction toggling, error handling, and existing word conflicts. These tests validate the complete user journey from API key setup through successful translations and error scenarios.

Also updates CI to run AI e2e tests separately on master/workflow_dispatch (not on PR gate) since they make real, billed calls to OpenAI.

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [x] Infrastructure / CI/CD
- [ ] Performance improvement

## Changes Made

- **New test files** (`app/src/androidTest/java/com/procrastilearn/app/e2e/ai/`):
  - `AiTranslationHappyPathE2eTest.kt` – API key setup, preview, and successful add/update flows with real OpenAI calls
  - `AiTranslationOfflineQueueE2eTest.kt` – Offline mode routes to pending queue instead of attempting translation
  - `AiTranslationDirectionE2eTest.kt` – Both translation directions (native→target and target→native) produce successful previews
  - `AiTranslationInvalidKeyE2eTest.kt` – Invalid API key surfaces error card instead of hanging
  - `AiTranslationExistingWordE2eTest.kt` – Existing word conflict with AI mode active triggers fresh translation
  - `AiE2eTestSupport.kt` – Shared test utilities (prefs seeding, database setup, UI helpers)

- **Build configuration** (`app/build.gradle.kts`):
  - Read `OPENAI_API_KEY` from environment or `local.properties`
  - Pass key to instrumented tests via runner arguments (never baked into APK)

- **CI workflow** (`.github/workflows/ci.yml`):
  - Exclude AI e2e tests from main instrumented test job (via `notPackage` filter)
  - Add separate `ai-e2e-tests` job that runs only on master/workflow_dispatch
  - Job sets up emulator, caches AVD, and runs AI tests with `OPENAI_API_KEY` secret

- **Minor UI changes** (`app/src/main/java/com/procrastilearn/app/ui/screens/AddWordFeedbackSections.kt`):
  - Add `testTag` to error card for test identification

- **DI export** (`app/src/main/java/com/procrastilearn/app/di/PreferencesEntryPoint.kt`):
  - Export `TranslationPreferences` for test access

## How to Test

**Local runs** (requires valid OpenAI API key):
1. Add `OPENAI_API_KEY=sk-...` to `local.properties`
2. Boot emulator: `emulator -avd Medium_Phone &`
3. Run all AI e2e tests: `./gradlew connectedDebugAndroidTest -Pandroid.testInstrumentationRunnerArguments.package=com.procrastilearn.app.e2e.ai`
4. Or run individual test class: `./gradlew connectedDebugAndroidTest --tests AiTranslationHappyPathE2eTest`

**CI**: Tests run automatically on push to master or via workflow_dispatch. Main instrumented test job excludes them to avoid billed API calls on every PR.

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases
- [x] Existing tests pass locally (main instrumented tests still pass with `notPackage` filter)
- [x] Updated documentation if needed (CI workflow comments explain the separation)
- [x] No duplication introduced (shared utilities in `AiE2eTestSupport.kt`)
- [x] No new warnings or suppressions introduced

## Related Issues

Validates the complete AI translation feature end-to-end with real API calls,

https://claude.ai/code/session_01RT3icJimphVb6aJKUAsNCm

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -143,3 +144,91 @@ jobs:
         with:
           name: instrumented-logcat
           path: /tmp/logcat.txt
+
+  ai-e2e-tests:
+    needs: verify
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    # Real, billed calls to the OpenAI API - kept off the PR gate. Runs after merge to master, or
+    # on demand via workflow_dispatch.
```
Comment (Vladyslav-Soldatenko) [inline .github/workflows/ci.yml]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiE2eTestSupport.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+private const val OPENAI_API_KEY_ARG = "OPENAI_API_KEY"
+
+/**
+ * A real OpenAI key is required: these tests hit the live API, and their whole point is that
+ * they fail clearly without a key and pass once one is provided (locally via local.properties,
+ * in CI via the OPENAI_API_KEY repository secret).
+ */
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiE2eTestSupport.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiE2eTestSupport.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+// The "use AI" Checkbox has no testTag/contentDescription and, whenever the bidirectional
+// checkbox is also visible (AI mode not active yet), isToggleable() alone isn't unique.
+fun ComposeTestRule.clickUseAiToggle(useAiToggleLabel: String) {
+    onNode(
+        isToggleable().and(hasAnySibling(hasText(useAiToggleLabel))),
+        useUnmergedTree = true,
+    ).performClick()
+}
+
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiE2eTestSupport.kt:101]: better add tag there to improve test code. Remmove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationDirectionE2eTest.kt [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Confirms an AI preview succeeds in both translation directions - a real, separate prompt path
+ * for each - via [com.procrastilearn.app.ui.AddWordViewModel.onTranslationDirectionToggle].
+ */
+@RunWith(AndroidJUnit4::class)
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationDirectionE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationDirectionE2eTest.kt [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        composeTestRule.onNodeWithText(string(R.string.add_word_preview_cancel)).performClick()
+        composeTestRule.waitForIdle()
+
+        composeTestRule
+            .onNodeWithContentDescription(string(R.string.add_word_toggle_direction))
+            .performClick()
+        composeTestRule.waitForIdle()
+
+        // onPreviewCancel clears the word field, so this is a fresh entry in the now-reversed
+        // direction (target -> native).
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationDirectionE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationExistingWordE2eTest.kt [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * The "word already exists, override?" conflict path (distinct from the Preview-triggered
+ * stored-translation/Regenerate path): tapping Add directly on a word that's already saved, with
+ * AI mode active, surfaces [com.procrastilearn.app.ui.screens.ExistingWordDialog] and proceeding
+ * makes a real AI call via [com.procrastilearn.app.ui.ExistingWordOverrideCoordinator].
+ */
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationExistingWordE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationHappyPathE2eTest.kt [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Real, end-to-end AI translation coverage: real Hilt DI, real Room database and a real call to
+ * the OpenAI API through the running app. Requires a valid OPENAI_API_KEY (see
+ * [requireOpenAiApiKey]) - JVM unit tests already cover the mocked/error-branch behavior of
+ * [com.procrastilearn.app.domain.usecase.GenerateAiTranslationUseCase].
+ */
+@RunWith(AndroidJUnit4::class)
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationHappyPathE2eTest.kt:26]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationHappyPathE2eTest.kt [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        composeTestRule.waitUntilNodeExists(hasContentDescription(viewListLabel), DEFAULT_TIMEOUT_MS)
+        composeTestRule.onNodeWithContentDescription(viewListLabel).performClick()
+        composeTestRule.waitForIdle()
+    }
+
+    private companion object {
+        const val DEFAULT_TIMEOUT_MS = 15_000L
+
+        // Real network call to OpenAI - generous timeout to avoid flaking on slow responses.
+        const val AI_CALL_TIMEOUT_MS = 30_000L
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationHappyPathE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationInvalidKeyE2eTest.kt [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * A real 401 from OpenAI, not a mocked one - no OPENAI_API_KEY secret is needed for this test
+ * since the key it seeds is deliberately invalid. The error text itself is whatever the OpenAI
+ * SDK's exception message says (not asserted on, since it isn't under this app's control); what
+ * matters is that the app surfaces [com.procrastilearn.app.ui.screens.ErrorMessageCard] instead
+ * of hanging or crashing.
+ */
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationInvalidKeyE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationOfflineQueueE2eTest.kt [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * No OpenAI API key/network is needed here - the point is that going offline with AI mode active
+ * routes Add through [com.procrastilearn.app.ui.AddWordViewModel.queuePendingWord] instead of
+ * attempting a translation call at all.
+ */
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationOfflineQueueE2eTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationOfflineQueueE2eTest.kt [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+    fun afterEach() {
+        setNetworkEnabled(true)
+        resetAiVocabulary(targetContext)
+        clearAiTranslationPrefs(targetContext)
+    }
+
+    @Test
+    fun addingAWordWhileOfflineQueuesItInsteadOfCallingAi() {
+        // A placeholder key is enough: AI mode only needs a non-blank key to activate, and the
+        // offline branch is taken before any translation call would be attempted.
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/ai/AiTranslationOfflineQueueE2eTest.kt]: remove comment

File: app/build.gradle.kts [hunk 12]
```diff
@@ -14,6 +16,19 @@ jacoco {
...
+val localProperties =
+  Properties().apply {
+    val file = rootProject.file("local.properties")
+    if (file.exists()) {
+      file.inputStream().use { load(it) }
+    }
+  }
+
+// Never a build-time secret (would end up baked into the compiled app/APK): passed only as an
+// instrumentation runner argument, read at test run time via InstrumentationRegistry.getArguments().
```
Comment (Vladyslav-Soldatenko) [inline app/build.gradle.kts]: remove comment
